<?xml version="1.0" encoding="utf-8"?>

<launch>
	<arg name="camera_enable" default="true"/>
	<arg name="camera_device" default="/dev/video0"/>
	<arg name="camera_info" default="file://$(find ar_sys)/camera_info/camera_calibration.yaml"/>
	<arg name="camera_frame_id" default="camera1"/>
	<arg name="camera_width" default="1280"/>
	<arg name="camera_height" default="1024" />
	<arg name="camera_fps"  default="30"/>

	<arg name="result_display" default="true"/>
	<arg name="result_autosize" default="true"/>
	<arg name="result_draw_markers" default="false" />
	<arg name="result_draw_markers_cube" default="false" />
	<arg name="result_draw_markers_axis" default="false" />

	<arg name="uid" default=""/>
	<arg name="video_namespace" default="/camera1"/>
	<arg name="video_image_topic" default="image_rect_color"/>
	<arg name="video_info_topic" default="camera_info"/>
	<arg name="video_rectified" default="true" />

	<arg name="board_config" default="$(find ar_sys)/data/single/board.yml" />
	<arg name="board_frame_id" default="board1" />
	<arg name="marker_size" default="0.05" />


	<group if="$(arg camera_enable)">
		<node ns="$(arg video_namespace)" pkg="image_proc" type="image_proc" name="image_proc$(arg uid)" />
		<node ns="$(arg video_namespace)" pkg="uvc_camera_node" type="uvc_camera_node" name="uvc_camera$(arg uid)" output="screen">
			<param name="width" type="int" value="$(arg camera_width)" />
			<param name="height" type="int" value="$(arg camera_height)" />
			<param name="fps" type="int" value="$(arg camera_fps)" />
			<param name="frame_id" type="string" value="/$(arg camera_frame_id)" />
			<param name="device" type="string" value="$(arg camera_device)" />
			<param name="camera_info_url" type="string" value="$(arg camera_info)" />
		</node>
	</group>

	<node ns="/" pkg="ar_sys" type="single_board" name="ar_single_board$(arg uid)" output="screen">
		<remap from="/camera_info" to="$(arg video_namespace)/$(arg video_info_topic)" />
		<remap from="/image" to="$(arg video_namespace)/$(arg video_image_topic)" />

		<param name="image_is_rectified" type="bool" value="$(arg video_rectified)"/>
		<param name="board_config" type="string" value="$(arg board_config)"/>
		<param name="board_frame" type="string" value="/$(arg board_frame_id)" />
		<param name="marker_size" type="string" value="$(arg marker_size)"/>
		<param name="draw_markers" type="bool" value="$(arg result_draw_markers)" />
		<param name="draw_markers_cube" type="bool" value="$(arg result_draw_markers_cube)" />
		<param name="draw_markers_axis" type="bool" value="$(arg result_draw_markers_axis)" />
	</node>

	<node ns="/" pkg="topic_tools" type="relay" name="ar_single_board_relay$(arg uid)" args="/ar_single_board$(arg uid)/transform /arsys_single_board/transform" />

	<group if="$(arg result_display)">
		<node ns="/" pkg="image_view" type="image_view" name="image_view$(arg uid)" output="screen">
			<remap from="image" to="/ar_single_board$(arg uid)/result" />

			<param name="autosize" type="bool" value="$(arg result_autosize)" />
		</node>
	</group>

</launch>
